File: 541-reverse-string-ii/reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {

        char[] array = s.toCharArray ();

        for ( int start = 0; start < array.length; start += 2*k)
        {
            int innerStart = start, innerEnd = Math.min(innerStart + k - 1, array.length - 1);

            while (innerStart < innerEnd)
            {
                char temp = array[innerStart];
                array[innerStart++] = array[innerEnd];
                array[innerEnd--] = temp;
            }
        }
        return new String (array);
    }
}

File: 541-reverse-string-ii/reverse-string-ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        s = list(s)

        for start in range (0, len(s), 2*k):
            innerStart, innerEnd = start, min(start + k - 1, len(s)-1)

            while innerStart < innerEnd:
                s[innerStart], s[innerEnd] = s[innerEnd],  s[innerStart]
                innerStart += 1
                innerEnd -= 1
        return "".join(s)